### PR TITLE
Create Dockerfile

### PR DESCRIPTION
**Pull request by tmcgilchrist**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM ocaml/opam:debian-11-ocaml-4.14@sha256:70bd0da74f68550d667cc1a8ae80661e694cd9ac614273b23c45f3fb8a9f3ac9 AS build
+RUN sudo apt-get update && sudo apt-get install libev-dev libffi-dev capnproto m4 pkg-config libsqlite3-dev libgmp-dev graphviz -y --no-install-recommends
+RUN cd ~/opam-repository && git pull origin -q master && git reset --hard ed966e5e82880cba7c483d3a5c1c8d4328c4f0c1 && opam update
```
Comment (MisterDA) [inline Dockerfile:3]: You can do a `fetch` instead of a `pull` since you're doing a hard reset afterwards

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+WORKDIR /src
+COPY --chown=opam solver-service.opam solver-service-api.opam \
+	solver-worker.opam /src/
+RUN opam-2.1 install -y --deps-only .
+ADD --chown=opam . .
+RUN opam-2.1 exec -- dune build ./_build/install/default/bin/solver-worker
```
Comment (MisterDA) [inline Dockerfile:9]: Would `opam-2.1 exec -- dune build ./bin/solver-worker` work instead?

File: Dockerfile (hunk 1 above)
Comment (tmcgilchrist) [reply Dockerfile:3]: Yes of course, this was copied from other Dockerfiles in ocluster that could use the same treatment.

File: Dockerfile (hunk 2 above)
Comment (tmcgilchrist) [reply Dockerfile:9]: This isn't the right path.

**Comment by moyodiallo** [comment]
This is the `Dockerfile` [here](https://github.com/moyodiallo/solver-service/blob/ocaml-multicore-ci-solver/Dockerfile), that's working in the live branch(https://deploy.ci3.ocamllabs.io/).

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM ocaml/opam:debian-11-ocaml-4.14@sha256:70bd0da74f68550d667cc1a8ae80661e694cd9ac614273b23c45f3fb8a9f3ac9 AS build
+RUN sudo apt-get update && sudo apt-get install libev-dev libffi-dev capnproto m4 pkg-config libsqlite3-dev libgmp-dev graphviz -y --no-install-recommends
+RUN cd ~/opam-repository && git fetch origin master && git reset --hard ed966e5e82880cba7c483d3a5c1c8d4328c4f0c1 && opam update
+WORKDIR /src
+COPY --chown=opam solver-service.opam solver-service-api.opam \
+	solver-worker.opam /src/
+RUN opam-2.1 install -y --deps-only .
+ADD --chown=opam . .
+RUN opam-2.1 exec -- dune build ./_build/install/default/bin/solver-worker
```
Comment (moyodiallo) [inline Dockerfile:9]: `solver-service` omitted here.
`RUN opam-2.1 exec -- dune build ./_build/install/default/bin/solver-service`

**Comment by tmcgilchrist** [comment]
Not required anymore.

**Pull request closed without merging**